### DISCUSSION TITLE
Which version of C++ required for translated code?

### DISCUSSION THREAD

**Discussion by ghost** [pfusik/cito#100] [Q&A] [answered]
C++20 is required for the transpiler `fut` itself. But what about normal code? C++20 is required, too?

**Comment by pfusik** [comment, marked as answer]
Depends on what you transpile. Some features transpile to C++20, but I think it's also possible that C++98 is enough.

**Reply by ghost** [reply]
Which features?

**Reply by pfusik** [reply]
This isn't a complete list, just some examples:
- dynamic references transpile to C++11 `std::shared_ptr`
- string references transpile to C++17 `std::string_view`
- string `StartsWith` transpiles to C++20 `starts_with`
- interpolated strings transpile to C++20 `std::format`
